src/Navigation/StackNavigation.js

```diff
@@ -8,21 +13,111 @@ import ProduceDetailsScreen from '../Marketplace/ProduceDetailsScreen';
 
 const stackNavig = createNativeStackNavigator();
 
-/* Stack navigators push/pop screens from a navigation stack (similar to web browser) */
-export default function StackNavigation() {
+const Drawer = createDrawerNavigator();
+
+const styles = StyleSheet.create({
+  icon: {
+    left: 25,
+  },
+});
+
+const profileIcon = ({ color }) => (
+  <Icon
+    size={26}
+    name="account-outline"
+    color={color}
+    style={styles.icon}
+  />
+);
+
+const marketplaceIcon = ({ color }) => (
+  <Icon
+    size={26}
+    name="store-outline"
+    color={color}
+    style={styles.icon}
+  />
+);
+
+const cartIcon = ({ color }) => (
+  <Icon
+    size={26}
+    name="cart-outline"
+    color={color}
+    style={styles.icon}
+  />
+);
+
+const contactIcon = ({ color }) => (
+  <Icon
+    size={26}
+    name="phone-outline"
+    color={color}
+    style={styles.icon}
+  />
+);
+
+function DrawerRoutes() {
   return (
-    <stackNavig.Navigator initialRouteName="SignIn">
-      <stackNavig.Screen name="SignIn" component={SignInScreen} />
-      <stackNavig.Screen name="Marketplace" component={MarketplaceScreen} />
-      <stackNavig.Screen name="Profile" component={ProfileScreen} />
-      <stackNavig.Screen name="Cart" component={CartScreen} />
-      <stackNavig.Screen
-        name="ProduceDetails"
-        component={ProduceDetailsScreen}
+    <Drawer.Navigator
+      initialRouteName="Back"
+      drawerContent={(props) => <CustomDrawer {...props} />}
+      screenOptions={{
+        drawerActiveBackgroundColor: '#FCF7F0',
+        drawerInactiveBackgroundColor: '#FCF7F0',
+        drawerActiveTintColor: '#34221D',
+        drawerInactiveTintColor: '#34221D',
+        drawerItemStyle: {
+          right: 25, borderTopRightRadius: 25, borderBottomRightRadius: 25, width: '80%',
+        },
+        drawerLabelStyle: { left: 10 },
+      }}
+    >
+      <Drawer.Screen
+        name="ProfileDrawer"
+        component={ProfileScreen}
+        options={{
+          title: 'Profile',
+          drawerIcon: profileIcon,
+        }}
+      />
+      <Drawer.Screen
+        name="Back"
+        component={MarketplaceScreen}
         options={{
-          title: '',
+          title: 'Marketplace',
+          drawerIcon: marketplaceIcon,
         }}
       />
+      <Drawer.Screen
+        name="Orders"
+        component={OrdersScreen}
```

Comment:
Since these haven't been merged yet, put MarketplaceScreen/some other screen that already exists in place of OrdersScreen and other missing ones and mark those lines w/ a TODO comment for future reference